outbox-core/src/main/java/io/praesidio/outbox/MessageRelay.java

```diff
@@ -17,26 +17,29 @@
 
     private final MessageRepository messageRepository;
     private final Map<MessageType, MessageRelayProvider> messageRelayProviders;
-    private final Map<MessageType, MessageSerializer> messageSerializers;
     private final TransactionValidator transactionValidator;
 
     public MessageRelay(
             MessageRepository messageRepository,
             Collection<MessageRelayProvider> messageRelayProviders,
-            Collection<MessageSerializer> messageSerializers,
             TransactionValidator transactionValidator
     ) {
         this.messageRepository = messageRepository;
         this.messageRelayProviders = messageRelayProviders.stream().collect(toMap(MessageRelayProvider::getType, identity()));
-        this.messageSerializers = messageSerializers.stream().collect(toMap(MessageSerializer::getType, identity()));
         this.transactionValidator = transactionValidator;
     }
 
+    public void relayMessage(MessageId messageId) {
+        messageRepository.findById(messageId)
+                         .ifPresent(this::relay);
+    }
+
     public void relayMessages() {
         messageRepository.findMessagesToRelay().forEach(this::relay);
     }
 
     private void relay(Message message) {
+        // FIXME possible double sending of a message - we should check if already sent
```

Comment:
Maybe a new issue or tests showing what can go wrong (if not too complex to write)?